projects/golang/go/development/docs/eks_go_patching_runbook.md

```diff
@@ -0,0 +1,145 @@
+
+For example, to obtain a SPEC file for Go `1.19`, do the following:
+1. Identify the proper RPM to base our EKS Go build on
+    1. identify the Amazon Linux 2 Golang RPM for the given version using `yum list —show-duplicates golang` on an Amazon Linux 2 machine.
+    1. If Amazon Linux does not yet support the given version of Golang, instead use [the Fedora yum repo](https://packages.fedoraproject.org/pkgs/golang/golang/) to find an appropriate RPM for the given Golang version.
+
+1. Download and unpack the target RPM:
+   1. download the target RPM from the package repo and unpack it. The command below will download the target RPM and unpack it into the `~/rpmbuild` directory.
+     The `spec` file will be output to `~/rpmbuild/SPECS` and any sources (such as test files and patches) will be
+     output to `~/rpmbuild/SOURCES`.
+   ```shell
+   yumdownloader --source golang-1.16.15-1.amzn2.0.1
+   rpm -ivh golang-1.16.15-1.amzn2.0.1.src.rpm
+   ```
+   2. Copy the source files from `~/rpmbuild/SOURCES` to `projects/golang/go/$VERSION/rpmbuild/SOURCES`
+   1. Copy the spec file from `~rpmbuild/SPECS` to `projects/golang/go/$VERSION/rpmbuild/SPECS`.
+
+1. Update the upstream Spec file with EKS Go conventions 
+   1. EKS Go uses a specific RPM Release string to identify itself. 
+Find [the `RELEASE` string in the unpacked spec file](https://github.com/aws/eks-distro-build-tooling/blob/main/projects/golang/go/1.18/rpmbuild/SPECS/golang.spec#L121)
+and modify it to use our format:
+   ```shell
+   Release:        %{?_buildid}%{?dist}.eks
+   ```
+   2. EKS Go builds from cloned upstream source at build-time, rather than a pre-packaged source included with the RPM. 
+      This means we need to modify the RPM setup directive to target the right directory. 
+      Modify [the `autosetup` directive to follow the pattern we use in our other RPM spec files.](https://github.com/aws/eks-distro-build-tooling/blob/main/projects/golang/go/1.19/rpmbuild/SPECS/golang.spec#L293) 
+      Specifically, ensure we're targeting the source directory of the format `go-go%{go_version}`.
+
+1. Add EKS Go specific patches
+
+   Next, add any additional required patch files to the `SOURCES` directory, and [update the modified spec file to apply them](https://github.com/aws/eks-distro-build-tooling/blob/main/projects/golang/go/1.19/rpmbuild/SPECS/golang.spec#L165).
+   Golang RPM spec files after Go 1.16 use the RPM directive [`autosetup`](https://github.com/aws/eks-distro-build-tooling/blob/main/projects/golang/go/1.19/rpmbuild/SPECS/golang.spec#L293), so all you need to do to apply a patch at build time is define it as a Patch source in the spec.
+   
+   EKS Go currently requires one custom patch for each Go version, above and beyond the security patches. 
+   This patch allows us to skip specific Golang standard library tests in certain circumstances, such as skipping privilleged tests which call `mount` when we're building the RPM in a container. 
```

Comment:
This could use a line with more context on why we skip those and if we should look out for changing that in the future if circumstances change

Reply:
added a footnote with the deets